bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/grpc/ChannelOptions.java

```diff
@@ -160,6 +164,9 @@ public ChannelOptions(Credentials credential,
       ScheduledExecutorService scheduledExecutorService,
       long timeoutMs,
       int channelCount) {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(userAgent),
```

Comment:
Some precondition checks are in the setter and others in the constructor. Consolidate?